src/tracker.hpp

```diff
@@ -23,71 +23,300 @@ See the Apache License Version 2.0 for the specific language governing permissio
 using std::string;
 using std::vector;
 
+/**
+ * @brief Singleton object that provides an interface to track Snowplow events.
+ */
 class Tracker {
 public:
+  /**
+   * @brief Initializes and returns the singleton tracker instance.
+   * 
+   * @param emitter The emitter to which events are sent (required).
+   * @param subject The user being tracked (optional).
+   * @param client_session Client session object responsible for tracking user sessions (optional). Attach a ClientSession context to each event.
+   * @param platform The platform the Tracker is running on, can be one of: web, mob, pc, app, srv, tv, cnsl, iot (defaults to srv).
+   * @param app_id Application ID (defaults to empty string).
+   * @param name_space The name of the tracker instance attached to every event (defaults to empty string).
+   * @param use_base64 Whether to enable base 64 encoding (defaults to true).
+   * @param desktop_context Whether to add a desktop_context, which gathers information about the device the tracker is running on, to each event (defaults to true).
+   * @return Tracker* 
+   */
   static Tracker *init(Emitter &emitter, Subject *subject, ClientSession *client_session, string *platform, 
     string *app_id, string *name_space, bool *use_base64, bool *desktop_context);
+
+  /**
+   * @brief Returns the initialized singleton tracker instance.
+   * 
+   * @return Tracker* Tracker instance.
+   */
   static Tracker *instance();
+
+  /**
+   * @brief Clean up the tracker to be used in main destructor.
+   */
   static void close();
 
+  /**
+   * @brief  Event to capture custom consumer interactions without the need to define a custom schema.
+   */
   class StructuredEvent {
   public:
+    /**
+     * @brief Name for the group of objects you want to track e.g. "media", "ecomm".
+     */
     string category; // required
+
+    /**
+     * @brief Defines the type of user interaction for the web object.
+     * 
+     * E.g., "play-video", "add-to-basket".
+     */
     string action; // required
+
+    /**
+     * @brief Identifies the specific object being actioned.
+     * 
+     * E.g., ID of the video being played, or the SKU or the product added-to-basket.
+     */
     string *label;
+  
+    /**
+     * @brief Describes the object or the action performed on it.
+     * 
+     * This might be the quantity of an item added to basket
+     */
     string *property;
+
+    /**
+     * @brief Quantifies or further describes the user action.
+     * 
+     * This might be the price of an item added-to-basket, or the starting time of the video where play was just pressed.
+     */
     double *value;
+
+    /**
+     * @brief Unix timestamp (in ms) when the event was created. Assigned automatically.
+     */
     unsigned long long timestamp;
+
+    /**
+     * @brief ID of the event (UUID v4) that is assigned automatically.
+     */
```

Comment:
Do you think it would make sense to mark this as deprecated. So they avoid to use it and in case they can raise an issue if they use it for something.
Unless, we intend to remove it directly in this release but in a different PR.

(same for `timestamp`)

Reply:
Good point! Added a deprecation warning for both.